src/datascience-ui/image-renderer/index.ts

```diff
@@ -0,0 +1,200 @@
+            if (imageMetadata?.height) {
+                img.height = imageMetadata.height;
+            }
+            if (imageMetadata?.width) {
+                img.width = imageMetadata.width;
+            }
+            if (imageMetadata?.unconfined === true) {
+                img.style.maxWidth = 'none';
+            }
+        } else {
+            const div = document.createElement('div');
+            div.onmouseover = onMouseOver;
+            div.onmouseout = onMouseOut;
+            div.style.backgroundColor = backgroundColor;
+            div.className = 'svgContent';
+            div.innerHTML = data as string;
+            container.appendChild(div);
+        }
+    } catch (ex) {
+        console.error(`Failed to render mime type ${mimeString}`, ex);
+    }
+}
+
+function createSaveAsButton(outputItem: OutputItem, ctx: RendererContext<unknown>) {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const btn = document.createElement('button') as any;
+    btn.style.position = 'absolute';
+    btn.style.top = '5px';
+    btn.style.left = '5px';
+    btn.className = 'plotIcon hidden';
+    btn.ariaPressed = 'false';
+    btn.title = 'Expand image';
+    btn.ariaLabel = 'Expand image';
+    btn.onclick = () => {
+        if (ctx.postMessage) {
+            ctx.postMessage({
+                type: 'saveAs',
+                outputId: outputItem.id,
+                mimeType: outputItem.mime
+            });
+        }
+    };
+    btn.innerHTML = `<span>
+                        <span className="image-button-child">
+                            <svg
+                                width="16"
+                                height="16"
+                                viewBox="0 0 16 16"
+                                fill="none"
+                                xmlns="http://www.w3.org/2000/svg"
+                            >
+                                <path
+                                    class='plotIconSvgPath'
+                                    d="M12.0147 2.8595L13.1397 3.9845L13.25 4.25V12.875L12.875 13.25H3.125L2.75 12.875V3.125L3.125 2.75H11.75L12.0147 2.8595ZM3.5 3.5V12.5H12.5V4.406L11.5947 3.5H10.25V6.5H5V3.5H3.5ZM8 3.5V5.75H9.5V3.5H8Z"
+                                />
+                            </svg>
+                        </span>
+                    </span>`;
+    return btn;
+}
+
+function createPlotViewerButton(outputItem: OutputItem, ctx: RendererContext<unknown>) {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const btn = document.createElement('button') as any;
+    btn.style.position = 'absolute';
+    btn.style.top = '5px';
+    btn.style.left = '45px';
+    btn.className = 'plotIcon hidden';
+    btn.ariaPressed = 'false';
+    btn.title = 'Save As';
+    btn.ariaLabel = 'Save As';
+    btn.onclick = () => {
+        if (ctx.postMessage) {
+            ctx.postMessage({
+                type: 'openPlot',
```

Comment:
One small tweak here that I think we should make. This is just a string to string comparison. Would be nice to hardcode it now that it's in the same project so we don't accidentally change one side and not the other. Also for SaveAs.

Reply:
Done